spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/TwoStepsConversions.java

```diff
@@ -205,11 +249,23 @@ public Value convertOnWrite(Object proppertyVal, DatastorePersistentProperty per
 		return writeConverter.apply(val);
 	}
 
-	private EntityValue convertOnWriteSingleEmbedded(Object val, String kindName) {
+	private EntityValue convertOnWriteSingleEmbedded(Object val, String kindName,
+			boolean isEmbeddedComponents) {
 		IncompleteKey key = this.objectToKeyFactory.getIncompleteKey(kindName);
 
 		FullEntity.Builder<IncompleteKey> builder = FullEntity.newBuilder(key);
-		this.datastoreEntityConverter.write(val, builder);
+		// Write an embedded Map
+		if (val instanceof Map) {
+			Map<String, ?> map = (Map<String, ?>) val;
+			for (String field : map.keySet()) {
+				builder.set(field, convertOnWrite(map.get(field),
+						isEmbeddedComponents, false, field));
```

Comment:
Move the entity annotation to the class level of the embedded thing so we resolve this depth problem. 

just check for the existing entity annotation on those sub things